app/src/pages/OnDeviceDisplay/ProtocolSetup/index.tsx

```diff
@@ -511,59 +531,67 @@ function PrepareToRun({
-          onClickSetupStep={() => {
-            setODDMaintenanceFlowInProgress()
-            launchLPC()
-          }}
-          title={t('labware_position_check')}
-          detail={t(
-            lpcDisabledReason != null ? 'currently_unavailable' : 'recommended'
-          )}
-          subDetail={
-            latestCurrentOffsets.length > 0
-              ? t('offsets_applied', { count: latestCurrentOffsets.length })
-              : null
-          }
-          status="general"
-          disabled={lpcDisabledReason != null}
-          disabledReason={lpcDisabledReason}
-        />
-        <ProtocolSetupStep
-          onClickSetupStep={() => setSetupScreen('labware')}
-          title={t('labware')}
-          detail={labwareDetail}
-          subDetail={labwareSubDetail}
-          status="general"
-          disabled={labwareDetail === null}
-        />
-        <ProtocolSetupStep
-          onClickSetupStep={() => setSetupScreen('liquids')}
-          title={t('liquids')}
-          status="general"
-          detail={
-            liquidsInProtocol.length > 0
-              ? t('initial_liquids_num', {
-                  count: liquidsInProtocol.length,
-                })
-              : t('liquids_not_in_setup')
-          }
-          disabled={liquidsInProtocol.length === 0}
-        />
+        {!isLoading ? (
+          <>
+            <ProtocolSetupStep
+              onClickSetupStep={() => setSetupScreen('instruments')}
+              title={t('instruments')}
+              detail={instrumentsDetail}
+              status={instrumentsStatus}
+              disabled={speccedInstrumentCount === 0}
+            />
+            <ProtocolSetupStep
+              onClickSetupStep={() => setSetupScreen('modules')}
+              title={t('modules')}
+              detail={modulesDetail}
+              status={modulesStatus}
+              disabled={protocolModulesInfo.length === 0}
+            />
+            <ProtocolSetupStep
+              onClickSetupStep={() => {
+                setODDMaintenanceFlowInProgress()
+                launchLPC()
+              }}
+              title={t('labware_position_check')}
+              detail={t(
+                lpcDisabledReason != null
+                  ? 'currently_unavailable'
+                  : 'recommended'
+              )}
+              subDetail={
+                latestCurrentOffsets.length > 0
+                  ? t('offsets_applied', { count: latestCurrentOffsets.length })
+                  : null
+              }
+              status="general"
+              disabled={lpcDisabledReason != null}
+              disabledReason={lpcDisabledReason}
+            />
+            <ProtocolSetupStep
+              onClickSetupStep={() => setSetupScreen('labware')}
+              title={t('labware')}
+              detail={labwareDetail}
+              subDetail={labwareSubDetail}
+              status="general"
+              disabled={labwareDetail === null}
```

Comment:
nit, better to check for if `labwareDetail` is undefined as well 

```suggestion
              disabled={labwareDetail == null}
```